src/FromJSONTrait.php

```diff
@@ -27,9 +27,38 @@ public static function fromJSON($jsonStr) {
 
         if (is_null($cfg)) {
             $err = json_last_error();
-            throw new \InvalidArgumentException("Invalid JSON: $err");
+            throw new JsonParseErrorException($cfg, $err, self::provideJsonErrorMessageForCode($err));
         }
         $called_class = get_called_class();
         return new $called_class($cfg);
     }
+
+    private static function provideJsonErrorMessageForCode($code)
+    {
+
+        /**
+         * This is a hastily thrown together fix for the fact that
+         * php 5.4 doesn't have json_last_error_msg(). I very much recommend
+         * that this is replaced with json_last_error_msg(), is removed and the minimum
+         * recommended version of php is changed to 5.5. Especially seeing as php 5.4
+         * is now no longer being supported as of me writing this.
+         * @see http://php.net/supported-versions.php
+         */
```

Comment:
I honestly think that doing this is fine seeing as the older versions of the library and the plugin will still be available and will work fine. To make it clearer the documentation can also be changed to show which version of the library is compatible with which php version.